### ISSUE TITLE
Give informative error message for mergeSequenceTable when rownames are NULL

### ISSUE THREAD

**Issue by mmcardozo** [benjjneb/dada2#721]
Hi,
I am trying to merge two sequencing tables from 2 runs one sample coming from Hiseq and 25 from Miseq and I get this error:

> seqtab <- mergeSequenceTables(St1, St2)
Error in rval[rownames(tab), colnames(tab)] + tab : 
  non-conformable arrays
Could you please help me to solve this error? 
Many thanks,
Magda

**Comment by benjjneb** [comment]
Can you provide the output of the following?

```
dim(St1); class(St1)
dim(St2); class(St2)
head(rownames(St1); head(colnames(St1)
head(rownames(St2); head(colnames(St2)
```

**Comment by mmcardozo** [comment]
>dim(St1)
[1]    25 10059

>class(St1)
[1] "matrix"

> dim(St2)
[1]   1 415

> class(St2)
[1] "matrix"

> head(rownames(St1))
[1] "1"  "10" "11" "12" "13" "14"

> head(rownames(St2))
NULL

> head(colnames(St1))
[1] "TAGGGAATTTTCTACAATGGACGAAAGTCTGATAGAGCAATGCCGCGTGAGGGATGAAGGTCTTCGGATCGTAAACCTCTGTTGTAAGGAAAGAACACACGCTTCAGGAAATGGTGGCGTGCTGACGGTACCTTACCAGAAAGTCACGGCTAACTATGTGCCAGCAGCCGCGGTAATACATAGGTGGCGAGCGTTATCCGGATTTATTGGGCGTAAAGAGTGCACAGGTGGTTTGTCAAGTCGATTGTTAAAGCCCGGGGCTCAACCCCGGTTCGCAGTCGAAACTGACAGACTAGAATATCGGTGGGAGTAGTAGAATTCGATGTGTAGCGGTGGAATGCATAAATATATCGAGGAATACCAATGGCGAAGGCAGCTACTTGAACGATTATTGACACTTAAGCACGAAAGCGTGGGGAGCAAATA"
[2] "TAGGGAATTTTCTACAATGGACGAAAGTCTGATAGAGCAATGCCGCGTGAGGGATGAAGGTCTTCGGATCGTAAACCTCTGTTGTAAGGAAAGAACACACGCTTCAGGAAATGGTGGCGTGCTGACGGTACCTTACCAGAAAGTCACGGCTAACTATGTGCCAGCAGCCGCGGTAATACATAGGTGGCGAGCGTTATCCGGATTTATTGGGCGTAAAGAGTGCACAGGTGGTTTGTCAAGTCGATTGTTAAAGCCCGGAGCTCAACTCCGGTTCGCAGTCGAAACTGACAGACTAGAATATCGGTGGGAGTAGTAGAATTCGATGTGTAGCGGTGGAATGCATAAATATATCGAGGAATACCAATGGCGAAGGCAGCTACTTGAACGATTATTGACACTTAAGCACGAAAGCGTGGGGAGCAAATA"
[3] "TCGAGAATAATTCACAATGGACGCAAGTCTGATGGTGCAACGCCGCGTGGAGGATGACGGTCTTAGGATTGTAAACTCCTGTCATCTGGGAGAAAGACCTGGCGATTAATAGTCGACAGGGTTGATAGTACCAGAAGAGGAAGGGACGGCTAACTTCGTGCCAGCAGCCGCGGTAATACGAAGGTCCCGAGCGTTATTCGGAATCACTGGGCGTAAAGGGAGCGTAGGCGGCGCGGTAAGTCGGATGTGAAATCCCGGGGCTCAACCCCGGAACTGCATCTGATACTGCCGTGCTAGAGTAATGGAGGGGTAGCTGGAATTCTCGGTGTAGCAGTGAAATGCGTAGATATCGAGAGGAACACTCGTGGCGAAAGCGAGCTACTGGACATTTACTGACGCTGAGGCTCGAAGGCTAGGGGAGCGAAAG"
[4] "TGAGGAATATTGGACAATGGAGGCAACTCTGATCCAGCCATGCCGCGTGCAGGAAGACTGCCCTATGGGTTGTAAACTGCTTTTATACAGGAAGAAACCGCTCTACGTGTAGAGCTCTGACGGTACTGTAAGAATAAGGATCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGATCCAAGCGTTATCCGGAATCATTGGGTTTAAAGGGTCCGTAGGTGGATAATTAAGTCAGAGGTGAAAGTTTGCAGCTCAACTGTAAAATTGCCTTTGAAACTGGTTATCTTGAGTTATTATGAAGTAGTTAGAATATGTAGTGTAGCGGTGAAATGCATAGATATTACATAGAATACCAATTGCGAAGGCAGATTACTAATAATCAACTGACACTGATGGACGAAAGCGTGGGGAGCGAACA"
[5] "TGGGGAATATTGCGCAATGGGCGAAAGCCTGACGCAGCCATGCCGCGTGTGTGAAGAAGGCCTTAGGGTTGTAAAGCACTTTCAATTGGGAAGAAAAGTTTGTCGTTAATAGCGCCAAGCTCTGACATTACCTTTAGAAGAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCGAGCGTTAATCGGAATTACTGGGCGTAAAGCGCGCGTAGGCGGTTTGTTAAGTCAGCTGTGAAAGCCCCGGGCTCAACCTGGGAACTGCAGTTGATACTGGCCGACTAGAGTACGAGAGAGGGAGGTAGAATTCCATGTGTAGCGGTGAAATGCGTAGATATATGGAGGAATACCAGTGGCGAAGGCGGCCTCCTGGCTCGATACTGACGCTGAGGTGCGAAAGCGTGGGGATCAAACA"
[6] "TGAGGAATATTGGGCAATGGAGGCAACTCTGACCCAGCCATGCCGCGTGCAGGAAGACGGCCCTATGGGTTGTAAACTGCTTTTATACAGGAAGAAACACTCGTACGTGTACGAGCTTGACGGTACTGTAAGAATAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTATCCGGAATCATTGGGTTTAAAGGGTCCGCAGGCGGTCAATTAAGTCAGAGGTGAAATACCATAGCTCAACTATGGAACTGCCTTTGATACTGGTTGACTTGAGTCATATGGAAGTAGATAGAATGTGTAGTGTAGCGGTGAAATGCATAGATATTACACAGAATACCGATTGCGAAGGCAGTCTACTACGTATGTACTGACGCTCATGGACGAAAGCGTGGGGAGCGAACA"

> head(colnames(St2))
[1] "TGAGGAATATTGCACAATGGAGGAAACTCTGATGCAGCAACGCCGCGTGGAGGATGACGCATTTCGGTGTGTAAACTCCTTTTATGAGTCAAGAAAATGACGGTAGCTCATGAATAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTATTCGGAATCACTGGGCGTAAAGGACGCGTAGGCGGGAAGCCAAGTCTGATGTGAAATCCTATGGCTCAACCATAGAACTGCATTGGAAACTGGTTACCTAGAGTATGGGAGGGGGAGATGGAATTAGTGGTGTAGGGGTAAAATCCGTAGATATCACTAGGAATACCGAAAGCGAAGGCGATCTCCTGGAACAATACTGACGCTAAGGCGTGAAAGCGTGGGGAGCAAACG"   
[2] "CCGCGGTAATACGGAGGGTGCAAGCGTTATTCGGAATCACTGGGCGTAAAGGACGCGTAGGCGGGAAGCCAAGTCTGATGTGAAATCCTATGGCTCAACCATAGAACTGCATTGGAAACTGGTTACCTAGAGTATGGGAGGGGGAGATGGAATTAGTGGTGTAGGGGTAAAATCCGTAGATATCACTAGGAATACCGAAAGCGAAGGCGATCTCCTGGAACAATACTGACGCTAAGGCGTGAAAGCGTGGGGAGCAAACG"                                                                                                                                                 
[3] "TGAGGAATATTGCACAATGGGCGAAAGCCTGATGCAGCAACGCCGCGTGGAGGATGACGCATTTCGGTGTGTAAACTCCTTTTATAAGGGAAGAAAATGACGGTACCTTATGAATAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTACTCGGAATCACTGGGCGTAAAGGACGCGTAGGCGGGTTGCCAAGTCAGATGTGAAATCCAATGGCTCAACCATTGAACTGCATTTGAAACTGGTAATCTAGAGTATGGGAGGGGGAGATGGAATTAGTAGTGTAGGGGTAAAATCCGTAGATATTACTAGGAATACCGAAAGCGAAGGCGATCTCCTGGAACATAACTGACGCTAAGGCGTGAAAGCGTGGGGAGCAAACA"   
[4] "TGGGGAATATTGGACAATGGACCAAAAGTCTGATCCAGCAATTCTGTGTGCACGATGAAGGTCTTCGGATTGTAAAGTGCTTTCAGGTGGGAAGAAGAAAGTGACGGTACCACCAGAAGAAGCGACGGCTAAATACGTGCCAGCAGCCGCGGTAATACGTATGTCGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGAGTCTAGGCGGCTTGTTAAGTCAGATGTGAAAATGCGGGGCTCAACTCCGTATTGCGTTTGAAACTGGCAGGCTAGAGTACTGGAGAGGTGGGCGGAACTACAAGTGTAGAGGTGAAATTCGTAGATATTTGTAGGAATGCCGATAGTGAAGACAGCTCACTGGACAGATACTGACGCTAAAGCTCGAAAGCGTGGGGAGCGAACA"
[5] "TGAGGAATATTGCACAATGGAGGAAACTCTGATGCAGCAACGCCGCGTGGAGGATGACGCATTTCGGTGTGTAAACTCCTTTTATATGTCAAGAAAATGACGGTAGCATATGAATAAGCACCGGCTAACTCCGTGCCAGCAGCCGCGGTAATACGGAGGGTGCAAGCGTTATTCGGAATCACTGGGCGTAAAGGACGCGTAGGCGGGATGCCAAGTCTGATGTGAAATCCTATGGCTCAACCATAGAACTGCATTGGAAACTGGTAACCTAGAGTATGGGAGGGGGAGATGGAATTAGTGGTGTAGGGGTAAAATCCGTAGATATCACTAGGAATACCGAAAGCGAAGGCGATCTCCTGGAACAATACTGACGCTAAGGCGTGAAAGCGTGGGGAGCAAACG"   
[6] "CCGCGGTAATACGTATGTCGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGAGTCTAGGCGGCTTGTTAAGTCAGATGTGAAAATGCGGGGCTCAACTCCGTATTGCGTTTGAAACTGGCAGGCTAGAGTACTGGAGAGGTGGGCGGAACTACAAGTGTAGAGGTGAAATTCGTAGATATTTGTAGGAATGCCGATAGTGAAGACAGCTCACTGGACAGATACTGACGCTAAAGCTCGAAAGCGTGGGGAGCGAACA" 

-----Original message-----

**Comment by benjjneb** [comment]
> > dim(St2)
> [1]   1 415
> > head(rownames(St2))
> NULL

This is the problem. Should there be only 1 sample in `St2`? Possibly if you give that sample a name (i.e. `rownames(St2) <- "some.sample.name") things will work correctly, but something else is wrong if you aren't expecting just a single sample in `St2`.

**Comment by mmcardozo** [comment]
Hi,



I gave a random number to the rowname of St2 rownames(St2) <- "99" (which was only one sample) and I was able to merge the tables.



Thank you





-----Original message-----

**Comment by benjjneb** [comment]
Great! Going to update this topic to an enhancement for a better error message when rownames are NULL.

**Comment by benjjneb** [comment]
Fixed in 0fb4d9efd0d2d3033ef2832235db3d77ac5c5a78
